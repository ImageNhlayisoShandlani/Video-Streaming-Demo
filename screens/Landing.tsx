import { Button } from "@react-navigation/elements";
import { useNavigation } from "@react-navigation/native";
import type { NativeStackNavigationProp } from '@react-navigation/native-stack';
import React from "react";
import { Dimensions, Image, ScrollView, Text, TouchableOpacity, View } from "react-native";
import { constants } from "../shared/constants";

type RootStackParamList = {
    Home: { catId: string };
    // add other screens here if needed
};

export default function Landing() {
    const navigation = useNavigation<NativeStackNavigationProp<RootStackParamList>>();

    function handleClick(category) {
        try {
            navigation.navigate('Home', { catId: category });
        } catch (e) {
            console.log(e)
        }
    }

    return (
        <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center', backgroundColor: constants.colors.mainBackground }}>

            <View style={{ flex: 1, width: Dimensions.get('screen').width }}>
                <ScrollView>
                    <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center', }}><Text style={{ fontSize: 20, marginTop: 10, color: '#eee', marginBottom: 10 }}>Choose you flavor</Text></View>
                    <View style={{ flex: 1, width: Dimensions.get('screen').width, flexWrap: 'wrap', flexDirection: 'row', justifyContent: 'space-between', position: 'relative', alignItems: 'stretch', padding: 10 }}>
                        <TouchableOpacity onPress={() => handleClick('Movies')} style={{ position: 'relative', overflow: 'hidden', borderRadius: 10, marginBottom: 15 }}>
                            <Image source={require('../assets/images/1.png')} style={{ width: 180, height: 180, position: 'absolute' }} />
                            <View style={{ width: 180, height: 180, alignItems: 'center', justifyContent: 'center', backgroundColor: 'rgba(0,0,0, 0.75)' }}>
                                <Text style={{ color: '#fff', fontSize: 18, fontWeight: '900', fontStyle: 'italic' }}>Movies</Text>
                            </View>
                        </TouchableOpacity>

                        <TouchableOpacity onPress={() => { }} style={{ position: 'relative', overflow: 'hidden', borderRadius: 10, marginBottom: 15 }}>
                            <Image source={require('../assets/images/1.png')} style={{ width: 180, height: 180, position: 'absolute' }} />
                            <View style={{ width: 180, height: 180, alignItems: 'center', justifyContent: 'center', backgroundColor: 'rgba(0,0,0, 0.75)' }}>
                                <Text style={{ color: '#fff', fontSize: 18, fontWeight: '900', fontStyle: 'italic' }}>Movies</Text>
                            </View>
                        </TouchableOpacity>

                        <TouchableOpacity onPress={() => { }} style={{ position: 'relative', overflow: 'hidden', borderRadius: 10, marginBottom: 15 }}>
                            <Image source={require('../assets/images/1.png')} style={{ width: 180, height: 180, position: 'absolute' }} />
                            <View style={{ width: 180, height: 180, alignItems: 'center', justifyContent: 'center', backgroundColor: 'rgba(0,0,0, 0.75)' }}>
                                <Text style={{ color: '#fff', fontSize: 18, fontWeight: '900', fontStyle: 'italic' }}>Movies</Text>
                            </View>
                        </TouchableOpacity>

                        <TouchableOpacity role="button" onPress={() => { }} style={{ position: 'relative', overflow: 'hidden', borderRadius: 10, marginBottom: 15 }}>
                            <Image source={require('../assets/images/splash-icon-dark.png')} style={{ width: 180, height: 180, position: 'absolute' }} />
                            <View style={{ width: 180, height: 180, alignItems: 'center', justifyContent: 'center', backgroundColor: 'rgba(0,0,0, 0.75)' }}>
                                <Text style={{ color: '#fff', fontSize: 18, fontWeight: '900', fontStyle: 'italic' }}>Movies</Text>
                            </View>
                        </TouchableOpacity>

                        <TouchableOpacity onPress={() => { }} style={{ position: 'relative', overflow: 'hidden', borderRadius: 10, marginBottom: 15 }}>
                            <Image source={require('../assets/images/1.png')} style={{ width: 180, height: 180, position: 'absolute' }} />
                            <View style={{ width: 180, height: 180, alignItems: 'center', justifyContent: 'center', backgroundColor: 'rgba(0,0,0, 0.75)' }}>
                                <Text style={{ color: '#fff', fontSize: 18, fontWeight: '900', fontStyle: 'italic' }}>Movies</Text>
                            </View>
                        </TouchableOpacity>
                    </View>
                </ScrollView>
            </View>
        </View>
    )
}