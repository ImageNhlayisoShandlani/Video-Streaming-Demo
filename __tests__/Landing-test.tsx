import React from "react";

import { render } from "@testing-library/react-native";

import Landing from "../screens/Landing";
import { NavigationContainer } from "@react-navigation/native";
import { createNativeStackNavigator } from "@react-navigation/native-stack";
import { TouchableOpacity, View, Image, Text } from "react-native";

const Stack = createNativeStackNavigator();
describe('<Landing/>', () => {

    test("Landing Screen Should Render the Correct Hedaer", () => {
        const { getByText, getAllByRole } = render(
            <NavigationContainer>

                <Landing />
            </NavigationContainer>
        );

        getByText("Choose you flavor");
        expect( getAllByRole('button')).toBeOnTheScreen();
    });

    test("Categroy card", () => {
        const tree = render(
        <TouchableOpacity onPress={() => {}} style={{ position: 'relative', overflow: 'hidden', borderRadius: 10, marginBottom: 15 }}>
            <Image source={require('../assets/images/1.png')} style={{ width: 180, height: 180, position: 'absolute' }} />
            <View style={{ width: 180, height: 180, alignItems: 'center', justifyContent: 'center', backgroundColor: 'rgba(0,0,0, 0.75)' }}>
                <Text>Sample Test</Text>
            </View>
        </TouchableOpacity>).toJSON();

        expect(tree).toMatchSnapshot();
    });
});