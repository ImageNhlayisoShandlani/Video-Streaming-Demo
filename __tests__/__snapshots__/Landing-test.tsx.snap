// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Landing/> Categroy card 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "borderRadius": 10,
      "marginBottom": 15,
      "opacity": 1,
      "overflow": "hidden",
      "position": "relative",
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../assets/images/1.png",
      }
    }
    style={
      {
        "height": 180,
        "position": "absolute",
        "width": 180,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0,0,0, 0.75)",
        "height": 180,
        "justifyContent": "center",
        "width": 180,
      }
    }
  >
    <Text>
      Sample Test
    </Text>
  </View>
</View>
`;
